app/src/main/java/zapsolutions/zap/lnurl/LnurlDecoder.java

```diff
@@ -0,0 +1,44 @@
+package zapsolutions.zap.lnurl;
+
+public class LnurlDecoder {
+
+    public static String decode(String lnurl) throws Exception {
+
+        if (!lnurl.substring(0, 5).toLowerCase().equals("lnurl")) {
```

Comment:
We should maybe also check for `null`, otherwise we just get a `NullPointerException` here.